File: spring project/animalshome/src/main/java/fmi/course/hcmi/animalshome/dto/PetAdDto.java
package fmi.course.hcmi.animalshome.dto;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import java.util.List;
import javax.validation.Valid;
import javax.validation.constraints.NotNull;

@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode
@ToString
@Getter
@Setter
public class PetAdDto {
    private Long id;

    @NotNull
    @Valid
    private PetDto petDto;

    private List<PhotoDto> photosDto;
}

File: frontend/Animals-Home/src/app/model/petAdDto.ts
import { PetDto } from './petDto';
import { PhotoDto } from './photoDto';

export class PetAdDto {
    id: number;
    petDto: PetDto;
    photosDto: Array<PhotoDto>;

   public setId(id: number){
        this.id = id;
   }

   public setPet(petDto: PetDto){
    this.petDto = petDto;
   }

    public setPhotos(photosDto: Array<PhotoDto>){
        this.photosDto = photosDto;
    }
}